python/ray/actor.py

```diff
@@ -850,7 +849,7 @@ def __ray_checkpoint_restore__(self):
             checkpoint_resumed = False
             if checkpoint_index is not None:
                 # Load the actor state from the checkpoint.
-                worker.actors[worker.actor_id] = (
+                worker.actors[worker.actor_id.id()] = (
```

Comment:
You can get rid of the .id() here (and also below), since ObjectIDs are hashable.